packages/twenty-shared/src/utils/validation/normalizeLocale.ts

```diff
@@ -0,0 +1,48 @@
+import { APP_LOCALES, SOURCE_LOCALE } from '@/translations';
+
+/**
+ * Maps language codes to full locale keys in APP_LOCALES
+ * Example: 'fr' -> 'fr-FR', 'en' -> 'en'
+ */
+const languageToLocaleMap = Object.keys(APP_LOCALES).reduce<
+  Record<string, string>
+>((map, locale) => {
+  // Extract the language code (part before the hyphen or the whole code if no hyphen)
+  const language = locale.split('-')[0].toLowerCase();
+
+  // Only add to the map if not already added or if the current locale is the source locale
+  // This ensures language codes map to their full locale version (e.g., 'es' -> 'es-ES')
+  // but preserves 'en' -> 'en' since it's the source locale
+  if (!map[language] || locale === SOURCE_LOCALE) {
+    map[language] = locale;
+  }
+
+  return map;
+}, {});
+
+/**
+ * Normalizes a locale string to match our supported formats
+ */
+export const normalizeLocale = (
+  value: string | null,
+): keyof typeof APP_LOCALES => {
+  if (value === null) return SOURCE_LOCALE;
```

Comment:
consistency with rest of codebase
```suggestion
  if (value === null) {
  return SOURCE_LOCALE;
  }
```

Reply:
Thanks